packages/edit-post/src/components/visual-editor/index.js

```diff
@@ -183,25 +206,59 @@ export default function VisualEditor( { styles } ) {
 
 	const blockSelectionClearerRef = useBlockSelectionClearer();
 
-	const layout = useMemo( () => {
+	// fallbackLayout is used if there is no Post Content,
+	// and for Post Title.
+	const fallbackLayout = useMemo( () => {
 		if ( isTemplateMode ) {
 			return { type: 'default' };
 		}
 
 		if ( themeSupportsLayout ) {
 			// We need to ensure support for wide and full alignments,
 			// so we add the constrained type.
-			return { ...defaultLayout, type: 'constrained' };
+			return { ...globalLayoutSettings, type: 'constrained' };
 		}
 		// Set default layout for classic themes so all alignments are supported.
 		return { type: 'default' };
-	}, [ isTemplateMode, themeSupportsLayout, defaultLayout ] );
+	}, [ isTemplateMode, themeSupportsLayout, globalLayoutSettings ] );
 
-	const blockListLayoutClass = classnames( {
-		'is-layout-constrained': themeSupportsLayout,
-		'is-layout-flow': ! themeSupportsLayout,
-		'has-global-padding': useRootPaddingAwareAlignments,
-	} );
+	const postContentBlock = useMemo( () => {
+		return findPostContent( parse( templateContent ) ) || {};
+	}, [ templateContent ] );
+
+	const postContentLayoutClasses = useLayoutClasses( postContentBlock );
+
+	const blockListLayoutClass = classnames(
+		{
+			'is-layout-flow': ! themeSupportsLayout,
+			'wp-container-visual-editor': themeSupportsLayout,
+		},
+		postContentLayoutClasses
+	);
+
+	const postContentLayoutStyles = useLayoutStyles(
+		postContentBlock,
+		'.wp-container-visual-editor'
```

Comment:
I'm wondering if this selector needs to be a higher specificity selector such as `.block-editor-block-list__layout.is-root-container`, which I think was used prior to this PR? If I use that one on this line, the layout styles appear to win out over the TwentyTwentyTwo styles as expected. But I wasn't sure if you changed this selector for another reason?

| Current PR (TwentyTwentyTwo's margins override the left content justification) | When updating to the higher specificity selector |
| --- | --- |
| ![image](https://user-images.githubusercontent.com/14988353/191433182-b2e48cd0-4f2d-4ba6-852c-fd170e53c54e.png) | ![image](https://user-images.githubusercontent.com/14988353/191432994-79fa6917-4d07-4f44-b26f-b2a374ab6f30.png) |

Reply:
Not really! Must be my subconscious drive to lower specificity everywhere 😅 
It's a good idea to use the selector we had before though.